pkg/controller/perconaservermongodbbackup/perconaservermongodbbackup_controller.go

```diff
@@ -131,31 +136,36 @@ func (r *ReconcilePerconaServerMongoDBBackup) Reconcile(ctx context.Context, req
 		return rr, errors.Wrap(err, "fields check")
 	}
 
-	cluster := &psmdbv1.PerconaServerMongoDB{}
+	cluster := new(psmdbv1.PerconaServerMongoDB)
 	err = r.client.Get(ctx, types.NamespacedName{Name: cr.Spec.GetClusterName(), Namespace: cr.Namespace}, cluster)
 	if err != nil {
-		return rr, errors.Wrapf(err, "get cluster %s/%s", cr.Namespace, cr.Spec.GetClusterName())
-	}
-
-	// TODO: Remove after 1.15
-	if cluster.CompareVersion("1.12.0") >= 0 && cr.Spec.ClusterName == "" {
-		cr.Spec.ClusterName = cr.Spec.PSMDBCluster
-		cr.Spec.PSMDBCluster = ""
-		if err := r.client.Update(ctx, cr); err != nil {
-			return rr, errors.Wrap(err, "failed to update clusterName")
+		if !k8serrors.IsNotFound(err) {
+			return rr, errors.Wrapf(err, "get cluster %s/%s", cr.Namespace, cr.Spec.GetClusterName())
 		}
+		cluster = nil
+		err = nil
```

Comment:
`cluster` won't be `nil`. I set it to nil because I don't want to compare `cluster` to empty struct below. I removed `err = nil`: https://github.com/percona/percona-server-mongodb-operator/pull/1001/commits/123a4e22b9cb9a034151bff5f86a54224f03b85c